### PR TITLE
chore: udpate dependencies

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
update deps, Fix #18856 

Lint is failing on dependabot PR

Screenshot 2024-10-14 at 8 34 36 AM

**Did you add tests for your changes?**
NA

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

File: lib/container/ContainerEntryModule.js [hunk 1]
```diff
@@ -230,7 +230,7 @@ class ContainerEntryModule extends Module {
 				`if (!${RuntimeGlobals.shareScopeMap}) return;`,
 				`var name = ${JSON.stringify(this._shareScope)}`,
 				`var oldScope = ${RuntimeGlobals.shareScopeMap}[name];`,
-				`if(oldScope && oldScope !== shareScope) throw new Error("Container initialization failed as it has already been initialized with a different share scope");`,
+				'if(oldScope && oldScope !== shareScope) throw new Error("Container initialization failed as it has already been initialized with a different share scope");',
```
Comment (snitin315) [inline lib/container/ContainerEntryModule.js:233]: Bugfix in [@stylistic/quotes rule](https://github.com/webpack/webpack/actions/runs/11320033480/job/31476957279?pr=18856) - https://github.com/eslint-stylistic/eslint-stylistic/pull/544

**Comment by alexander-akait** [comment]
@snitin315 Looks like `rimraf` s broken 

```js
$ yarn mkdirp .nyc_output && nyc merge .nyc_output coverage/coverage-nyc.json && rimraf .nyc_output
$ /home/vsts/work/1/s/node_modules/.bin/mkdirp .nyc_output
/home/vsts/work/1/s/node_modules/foreground-child/dist/commonjs/index.js:13
const spawn = process?.platform === 'win32' ? cross_spawn_1.default : child_process_1.spawn;
```

we can install old version

**Comment by snitin315** [comment]
@alexander-akait where do you see this? It is working fine locally for me, also we have not updated `rimraf` in this PR

**Comment by alexander-akait** [comment]
@snitin315 

> webpack.webpack (Linux node-10-a) 

Azure

**Comment by snitin315** [comment]
@alexander-akait Fixed.

**Review by alexander-akait — APPROVED** [review]

**Pull request merged**